src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,333 @@
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerFactory.FIXED_LENGTH_ALGORITHM;
+
+/**
+ * This processor is used for chunking user input data text embedding processing, algorithm can be used to indicate chunking algorithm and parameters,
+ * and field_map can be used to indicate which fields needs chunking and the corresponding keys for the chunking results.
+ */
+@Log4j2
+public final class DocumentChunkingProcessor extends AbstractProcessor {
+
+    public static final String TYPE = "chunking";
+
+    public static final String FIELD_MAP_FIELD = "field_map";
+
+    public static final String ALGORITHM_FIELD = "algorithm";
+
+    public static String MAX_CHUNK_LIMIT_FIELD = "max_chunk_limit";
+
+    private static final int DEFAULT_MAX_CHUNK_LIMIT = -1;
+
+    private int current_chunk_count = 0;
+
+    private int max_chunk_limit = DEFAULT_MAX_CHUNK_LIMIT;
+    private final Set<String> supportedChunkers = ChunkerFactory.getAllChunkers();
+
+    private String chunkerType;
+
+    private Map<String, Object> chunkerParameters;
+
+    private final Map<String, Object> fieldMap;
+
+    private final ClusterService clusterService;
+
+    private final IndicesService indicesService;
+
+    private final AnalysisRegistry analysisRegistry;
+
+    private final Environment environment;
+
+    public DocumentChunkingProcessor(
+        String tag,
+        String description,
+        Map<String, Object> fieldMap,
+        Map<String, Object> algorithmMap,
+        Environment environment,
+        ClusterService clusterService,
+        IndicesService indicesService,
+        AnalysisRegistry analysisRegistry
+    ) {
+        super(tag, description);
+        validateAndParseAlgorithmMap(algorithmMap);
+        this.fieldMap = fieldMap;
+        this.environment = environment;
+        this.clusterService = clusterService;
+        this.indicesService = indicesService;
+        this.analysisRegistry = analysisRegistry;
+    }
+
+    public String getType() {
+        return TYPE;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void validateAndParseAlgorithmMap(Map<String, Object> algorithmMap) {
+        if (algorithmMap.size() != 1) {
+            throw new IllegalArgumentException(
+                "Unable to create the processor as [" + ALGORITHM_FIELD + "] must contain and only contain 1 algorithm"
+            );
+        }
+
+        for (Map.Entry<String, Object> algorithmEntry : algorithmMap.entrySet()) {
+            String algorithmKey = algorithmEntry.getKey();
+            Object algorithmValue = algorithmEntry.getValue();
+            if (!supportedChunkers.contains(algorithmKey)) {
+                throw new IllegalArgumentException(
+                    "Unable to create the processor as chunker algorithm ["
+                        + algorithmKey
+                        + "] is not supported. Supported chunkers types are ["
+                        + FIXED_LENGTH_ALGORITHM
```

Comment:
Use a function to build the supported chunk algorithms so that this can adapt to changes in the future.

Reply:
Already changed.